qubesadmin/tools/__init__.py

```diff
@@ -171,11 +181,25 @@ def parse_qubes_app(self, parser, namespace):
                 vm_name = getattr(namespace, self.dest, None)
                 if vm_name is not None:
                     try:
-                        namespace.domains += [app.domains[vm_name]]
+                        if re.match(UUID_REGEX, vm_name):
+                            namespace.domains += [uuid_to_domain(vm_name[5:])]
+                        else:
+                            namespace.domains += [app.domains[vm_name]]
                     except KeyError:
                         parser.error('no such domain: {!r}'.format(vm_name))
             else:
-                for vm_name in getattr(namespace, self.dest):
+                destinations=set()
+                for destination in getattr(namespace, self.dest, None):
+                    if any(wildcard in destination for wildcard in '*![]'):
+                        for domain in app.domains:
+                            if fnmatch.fnmatch(domain.name, destination):
+                                destinations.add(domain.name)
+                    elif re.match(UUID_REGEX, destination):
```

Comment:
With UUID-aware VMCollection, this case won't be necessary

Reply:
overall it is necessary to find out why `qubesadmin.Qubes().domains["uuid:..."]` does not work.